server/src/lib/api.ts

```diff
@@ -279,15 +279,16 @@ export default class API {
         break;
 
       case 'file':
-        avatarURL =
-          'data:' +
-          headers['content-type'] +
-          ';base64,' +
-          body.toString('base64');
-        console.log(avatarURL.length);
-        if (avatarURL.length > 8000) {
-          error = 'too_large';
-        }
+        let fileName = MD5(user.emails[0].value).toString() + '-avatar.png';
```

Comment:
So you wouldn't have known this off-hand, but the way our S3 is structured, each contributor already has a unique file folder at `BUCKET/client_id/` where all of their clips are stored. We picked this because we didn't always use emails as logins, and also so that people can change their email addresses without affecting their past contributions. While this implementation is probably fine since email changes are associated in the db and the avatar URL is still stored in the DB, for the sake of cleanliness I think it would make more sense for the filename to be `${client_id}/avatar`.

Also, I haven't dug too deeply into what's happening in `avatar-setup.tsx`, but I don't think avatars default to PNGs.

Reply:
Yes, I followed the Auth0 instructions from the readme. I will make the changes you requested.

Reply:
Just a follow-up question: will the request (sent to saveAvatar) always have the client_id property?